org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java

```diff
@@ -82,8 +83,24 @@ public void visitEnd() {
 		};
 	}
 
+	/**
+	 * @return <code>true</code> if method should not be analyzed
+	 */
 	// TODO: Use filter hook in future
-	private boolean isMethodFiltered(final int access, final String name) {
+	private boolean isMethodFiltered(final String className,
+			final String superClassName, final int access, final String name,
+			final String desc) {
+		if ("java/lang/Enum".equals(superClassName)) {
+			// filter out methods that compiler creates for enums
+			if ("values".equals(name)
```

Comment:
Umm, below there is anyway `name.startsWith(...)` that isn't NPE safe. Also according to my knowledge of ASM API - `desc` can't be `null`. For the NPE safety IMO far better to ensure that parameters of methods are not `null` by default and nullability should be stated explicitly.

But ok, I'll swap arguments, hoping that  you won't say that concatenation on left hand side looks awkward 😆